internal/msalbase/authentication_result.go

```diff
@@ -5,82 +5,67 @@ package msalbase
-		expiresOn, err = ConvertStrUnixToUTCTime(storageTokenResponse.accessToken.GetExpiresOn())
-		if err != nil {
-			return nil, errors.New("access token in cache expires at an invalid time")
-		}
-		grantedScopes = SplitScopes(storageTokenResponse.accessToken.GetScopes())
-	} else {
-		return nil, errors.New("no access token present in cache")
+	accessToken := storageTokenResponse.AccessToken.GetSecret()
+	expiresOn, err := ConvertStrUnixToUTCTime(storageTokenResponse.AccessToken.GetExpiresOn())
+	if err != nil {
+		return AuthenticationResult{}, errors.New("access token in cache expires at an invalid time")
 	}
-	// Checking if there was an ID token in the cache; this will throw an error in the case of confidential client applications
-	if !reflect.ValueOf(storageTokenResponse.idToken).IsNil() {
-		idToken, err = CreateIDToken(storageTokenResponse.idToken.GetSecret())
+	grantedScopes := SplitScopes(storageTokenResponse.AccessToken.GetScopes())
+
+	// Checking if there was an ID token in the cache; this will throw an error in the case of confidential client applications.
+	var idToken IDToken
+	if storageTokenResponse.IDToken != nil {
+		idToken, err = CreateIDToken(storageTokenResponse.IDToken.GetSecret())
 		if err != nil {
-			return nil, err
+			return AuthenticationResult{}, err
 		}
 	}
-	ar := &AuthenticationResult{account, idToken, accessToken, expiresOn, grantedScopes, declinedScopes}
-	return ar, nil
+	return AuthenticationResult{account, idToken, accessToken, expiresOn, grantedScopes, nil}, nil
 }
 
-// CreateAuthenticationResult creates an AuthenticationResult
-func CreateAuthenticationResult(tokenResponse *TokenResponse, account *Account) (*AuthenticationResult, error) {
-	grantedScopes := tokenResponse.GrantedScopes
-	declinedScopes := tokenResponse.declinedScopes
-	//Checking if there are any scopes that were declined, this would lead to an invalid access token
-	if len(declinedScopes) > 0 {
-		return nil, errors.New("token response failed because declined scopes are present")
+// CreateAuthenticationResult creates an AuthenticationResult.
+// TODO(jdoak): Make this a method on TokenResponse() that takes only 1 arge, Account.
+func CreateAuthenticationResult(tokenResponse TokenResponse, account Account) (AuthenticationResult, error) {
+	if len(tokenResponse.declinedScopes) > 0 {
+		return AuthenticationResult{}, errors.New("token response failed because declined scopes are present")
 	}
-	idToken := tokenResponse.IDToken
-	accessToken := tokenResponse.AccessToken
-	expiresOn := tokenResponse.ExpiresOn
-	ar := &AuthenticationResult{account, idToken, accessToken, expiresOn, grantedScopes, declinedScopes}
-	return ar, nil
+	return AuthenticationResult{
+		Account:        account,
+		idToken:        tokenResponse.IDToken,
+		AccessToken:    tokenResponse.AccessToken,
+		ExpiresOn:      tokenResponse.ExpiresOn,
+		GrantedScopes:  tokenResponse.GrantedScopes,
+		DeclinedScopes: nil,
```

Comment:
nit: not sure what's the point of exposing this field if it's always null.

Reply:
This is another example of github needing a better diff engine.  It's hiding what was really changed:
Original code:

```go
// CreateAuthenticationResult creates an AuthenticationResult
func CreateAuthenticationResult(tokenResponse *TokenResponse, account *Account) (*AuthenticationResult, error) {
	grantedScopes := tokenResponse.GrantedScopes
	declinedScopes := tokenResponse.declinedScopes
	//Checking if there are any scopes that were declined, this would lead to an invalid access token
	if len(declinedScopes) > 0 {
		return nil, errors.New("token response failed because declined scopes are present")
	}
	idToken := tokenResponse.IDToken
	accessToken := tokenResponse.AccessToken
	expiresOn := tokenResponse.ExpiresOn
	ar := &AuthenticationResult{account, idToken, accessToken, expiresOn, grantedScopes, declinedScopes}
	return ar, nil
}
```
This was returning the equivalent of a nil declinded scopes when it did its check, by returning a nil value.  You will never get that field filled in the original code.  

I didn't remove it, because I wasn't sure if that field can be set somewhere else.  If not, then I agree that it should not just unexposed, it should just be eliminated.  

Let me know what you want me to do here.

Reply:
This is fine, DeclinedScopes is part of the public API so we should not remove it. As long as developers understand which scopes were declined, it's fine by me.

For context - "Declined scopes" is a feature where MSAL libraries haven't fully aligned. Many MSAL libs **ignore** declined scopes. This is pretty bad, as it causes cache misses, but it's hard to change a default without incrementing major version lib, which is hard for a miriad of reasons (docs, samples, partners etc.). 
New libraries SHOULD fail when there are declined scopes, because the access token is not usable. The developer must be informed of which scopes were not issued.

Reply:
@bgavrilMS 

The explanation makes sense.  

If DeclindedScopes will never be set for this type of result, it should be removed.  Its just cruft and I don't think consistency where two APIs work differently due to language, etc... is a good reason to keep it.  

I would like to remove this, objections?